packages/@tinacms/graphql/src/primitives/resolver/index.ts

```diff
@@ -195,87 +202,169 @@ export class Resolver {
     }
   }
 
-  public resolveDocument = async ({
```

Comment:
The diff on `resolveDocument()` is pretty messy.  Basically, `resolveDocument()` is responsible for getting the `collection` figured out and the `realPath` (which requires the `collection` to know the `collection.path`).

Everything else is handled by a series of flags routing to either `createResolveDocument()` or `updateResolveDocument()` (not great names, but at least they won't clash easily).  The contents of `isAddPendingPage` is straight up lifted out of `resolveDocument()` with our new way of writing a document tucked underneath.

Everything _eventually_ returns a `this.getDocument(realPath)`.